tests/unit/view/test_dynamic_view.py

```diff
@@ -0,0 +1,420 @@
+        r2 = empty_view.add(b, c)
+        r3 = empty_view.add(c, e)
+    with empty_view.parallel_sequence(True):
+        r4 = empty_view.add(b, d)
+        r5 = empty_view.add(d, e)
+    r6 = empty_view.add(e, f)
+
+    assert r1.order == "1"
+    assert r2.order == "2"
+    assert r3.order == "3"
+    assert r4.order == "2"
+    assert r5.order == "3"
+    assert r6.order == "4"
+
+
+def test_hydration(empty_model: Model):
+    """Check dehydrating and hydrating."""
+    system = empty_model.add_software_system(name="system", id="sys1")
+
+    view = DynamicView(key="dyn1", description="Description", software_system=system)
+    view.set_model(empty_model)
+
+    io = DynamicViewIO.from_orm(view)
+    d = io.dict()
+    assert d["elementId"] == "sys1"
+
+    view2 = DynamicView.hydrate(io, element=system)
+    assert view2.key == "dyn1"
+    assert view2.description == "Description"
+    assert view2.element is system
+
+
+def test_relationships_are_ordered(empty_model: Model):
+    """Check that relationships are ordered for DynamicView."""
+    system1 = empty_model.add_software_system(name="System 1")
+    container1 = system1.add_container(name="Container 1")
+    container2 = system1.add_container(name="Container 2")
+    container1.uses(container2)
+
+    view = DynamicView(key="dyn1", description="test", software_system=system1)
+    view.set_model(empty_model)
+
+    # Test using 10 items, so we can check 1 < 2 < 10 (i.e. not string ordering)
+    for i in range(10):
+        view.add(container1, container2, description=f"rel {i}")
+
+    relationship_views = list(view.relationship_views)
+
+    assert len(relationship_views) == 10
+    for i in range(10):
+        assert relationship_views[i].order == str(i + 1)
+
+
+def test_relationships_with_subsequences_are_ordered(empty_model: Model):
+    """Test ordering works with subsequences."""
+    system1 = empty_model.add_software_system(name="System 1")
+    container1 = system1.add_container(name="Container 1")
+    container2 = system1.add_container(name="Container 2")
+    container1.uses(container2)
+
+    view = DynamicView(key="dyn1", description="test", software_system=system1)
+    view.set_model(empty_model)
+
+    view.add(container1, container2, description="test 1")
+    with view.subsequence():
+        view.add(container1, container2, description="test 1.1")
+        view.add(container1, container2, description="test 1.2")
+    for i in range(2, 11):
+        view.add(container1, container2, description=f"test {i}")
+
+    relationship_views = list(view.relationship_views)
+
+    assert relationship_views[0].order == "1"
+    assert relationship_views[1].order == "1.1"
+    assert relationship_views[2].order == "1.2"
+    assert relationship_views[3].order == "2"
+    assert relationship_views[11].order == "10"
+
+
+def test_relationship_view_sorting():
+    """Check various cases for sorting relationship views into order."""
```

Comment:
Would love to see some cases that test sorting, for example, '1.1' < '1.10' < '1.11' < '10.1' < '10.10'

Reply:
Done now in `test_view_order.py`.